web/src/users/add_url.rs

```diff
@@ -0,0 +1,41 @@
+use axum::{
+    extract::{Form, State},
+    response::{IntoResponse, Redirect},
+};
+use cja::jobs::Job;
+
+use crate::{jobs::process_article::ProcessArticle, AppState};
+
+use db::{urls::add_url, users::User};
+use serde::Deserialize;
+
+#[derive(Deserialize)]
+pub struct ArticleForm {
+    url: String,
+}
+
+#[axum::debug_handler(state = AppState)]
+pub async fn insert_article_handler(
+    State(state): State<AppState>,
+    user: User,
+    Form(form): Form<ArticleForm>,
+) -> impl IntoResponse {
+    let url = form.url;
+    let user_id = user.user_id;
+
+    match add_url(&state.db, &url, user_id, &true).await {
+        Ok(page) => {
+            let page_id = page.page().page_id;
+            let process_article = ProcessArticle { page_id };
+            process_article
+                .enqueue(state.clone(), "context".to_string())
```

Comment:
Ahh ya I never really explained what `context` should be!

So `context` is basically just a way for us a devs to be able to trace these jobs better.
In a big app we may be proccessing articles from all over the place. Like one might be this HTML form, and another might be from a Slack/Discord integration (that's likely coming up soonish 😉 )
And it's nice to be able to know 'where' jobs are coming from when are looking into the system!

So for this context... Lets use... `insert_article_handler` I think just to match the function name.
Another alternative is something more like `POST /articles` to match the route... 🤷 
No strong feels, but lets def make this more about the context of what we are doing/where we are in code!